### PR TITLE
[IMP] mail: Allow overwriting field defaults with mail data model.

### PR DESCRIPTION
**Pull request by elmeriniemela**
We could not find a way to implement the following modification:

We would like to fix this with code like this:

```javascript
odoo.define('sprintit_chatter_clear_checkbox/static/src/models/chatter/chatter.js', function (require) {
    'use strict';
    const { attr } = require('mail/static/src/model/model_field.js');
    const { registerFieldPatchModel } = require('mail/static/src/model/model_core.js');
    registerFieldPatchModel('mail.suggested_recipient_info', 'sprintit_chatter_clear_checkbox', {
        isSelected: attr({
            default: false,
        }),
    });
});

```

Current behavior before PR:

The code above does not work, the checkbox is still ticked.

Desired behavior after PR is merged:

After this pull request, the code above works and now the checkbox is **_not_** ticked by default.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/model/model_manager.js [hunk 1]
```diff
@@ -269,6 +269,7 @@ class ModelManager {
                 Model.fields[fieldName] = field;
             } else {
                 Object.assign(Model.fields[fieldName].dependencies, field.dependencies);
+                Model.fields[fieldName].default = field.default;
```
Comment (seb-odoo) [inline addons/mail/static/src/model/model_manager.js:272]: Ok for the idea, but in this case the original `default` would be overwritten if `field.default` is `undefined` which is an issue (it means every field patch will clear existing default which is not what we want).

Also note (in regard to your commit tile) that this change is not related to OWL in any way, it's just related to how we manage the data in this module.

Comment (elmeriniemela) [reply addons/mail/static/src/model/model_manager.js:272]: Ah yes you're right, so is it OK if I add a condition for the undefined case? And if there is some way to do this extension without any changes to Odoo code, I'm also interested in hearing that. We just need this modification implemented somehow.

Comment (seb-odoo) [reply addons/mail/static/src/model/model_manager.js:272]: In your case the `default` is not very useful anyway because it is on a field that is also computed, so you could just override `_computeIsSelected` with `registerInstancePatchModel` and return what you need.

Comment (seb-odoo) [reply addons/mail/static/src/model/model_manager.js:272]: If the user still needs to be able to change the value without the compute resetting it back though, you'll probably need to do it differently.

I guess in that case overriding `_created` to update the value to false once after creation would be preferable.

Comment (elmeriniemela) [reply addons/mail/static/src/model/model_manager.js]: We implemented the modification with `_created`, thanks alot for the tip!

You can close this if you think using `_created` is the preffered way.

Comment (seb-odoo) [reply addons/mail/static/src/model/model_manager.js]: Ok, well as it is safer to avoid changing code in stable unless absolutely necessary I will close this PR.

But in the future I'll keep in mind that it would be nice to be able to override the declaration of all field options and not just dependencies.
